### DISCUSSION TITLE
Items not loading in Chrome extension 2024.5.0

### DISCUSSION THREAD

**Discussion by bunicb** [dani-garcia/vaultwarden#4599] [Q&A]
### Subject of the issue
Items not loading in the Chrome extension after updating to version 2024.5.0. Manually installing extension 2024.4.2 provides all items from vault.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.30.5
* Web-vault version: v2024.1.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.44.0
* Clients used: Chrome extension, android application
* Reverse proxy and version: nginx-proxy-manager 2.10.4
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** SIGNUPS_ALLOWED, ADMIN_TOKEN

```json
{
  "_duo_akey": "***",
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**************",
  "domain_origin": "*****://**************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": true,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "*****************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "********************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Steps to reproduce
Update extension to 2024.5.0, login.

### Expected behaviour
Items load normally.

### Actual behaviour
No items load, spinning animation going on forever.

### Troubleshooting data
Checked if there are any reported issues on Bitwarden github for clients, but none were apparent for the newest version.

**Comment by BlackDex** [comment]
Seems to work just fine for me.
Do you see any connection attempts and/or errors why you try to sync via the Chrome Extension?

**Comment by bunicb** [comment]
Sync seems to work fine, I get the green banner with Syncing complete.
When looking in the Chrome console I get one error periodically:
```
background.js:2 Uncaught (in promise) TypeError: Failed to execute 'decrypt' on 'SubtleCrypto': The provided value is not of type '(ArrayBuffer or ArrayBufferView)'.
    at ip.<anonymous> (background.js:2:942102)
    at Generator.next (<anonymous>)
    at s (background.js:2:936910)
```
The next one is shown when switching to vault tab, and has a larger stack trace:
```
Unhandled error in angular Error: Uncaught (in promise): TypeError: Failed to execute 'decrypt' on 'SubtleCrypto': The provided value is not of type '(ArrayBuffer or ArrayBufferView)'.
TypeError: Failed to execute 'decrypt' on 'SubtleCrypto': The provided value is not of type '(ArrayBuffer or ArrayBufferView)'.
    at NV.<anonymous> (web-crypto-function.service.ts:342:38)
    at Generator.next (<anonymous>)
    at o (main.js:10:348918)
    at u.invoke (zone.js:368:26)
    at Object.onInvoke (ng_zone.ts:443:29)
    at u.invoke (zone.js:367:52)
    at s.run (zone.js:129:43)
    at zone.js:1257:36
    at u.invokeTask (zone.js:402:31)
    at Object.onInvokeTask (ng_zone.ts:428:29)
    at I (zone.js:1193:31)
    at zone.js:1100:17
    at zone.js:1116:33
    at a (main.js:1:1146473)
    at u.invoke (zone.js:368:26)
    at Object.onInvoke (ng_zone.ts:443:29)
    at u.invoke (zone.js:367:52)
    at s.run (zone.js:129:43)
    at zone.js:1257:36
    at u.invokeTask (zone.js:402:31)
```

I tried bypassing the reverse proxy, but to no avail.

**Comment by BlackDex** [comment]
Try to fully logout and login and see what that does.

**Comment by bunicb** [comment]
Tried it, same issue. Tried logging in to Bitwarden cloud, just to confirm it's doing this only with my instance, and that worked without issues.
Desktop app of the same version seems to work fine.

**Comment by BlackDex** [comment]
Now, i do not think it would do that much, but could you try the `testing` tagged Vaultwarden image? And see if that somehow solves it?

**Comment by BlackDex** [comment]
Not sure, but could you check if you might have some invalid data in the database by running the following Database Query
**after making a backup of the database**:
```sql
SELECT * FROM ciphers WHERE atype=2 AND data LIKE '%"Type":null%';
```

If that returns one or more rows, then try to execute this query:

```sql
UPDATE ciphers
SET data = REPLACE(data, '"Type":null', '"Type":0')
WHERE data LIKE '%"Type":null%';
```

**Comment by bunicb** [comment]
Tried the testing image, no difference.
The SELECT query returned zero rows.

**Comment by LiloBzH** [comment]
Same issue with Firefox Extension

sometimes , after sleep of windows, 
after entering the password, the extension starts loading but no longer moves.
To unblock, you must restart Firefox or deactivate then reactivate the extension (and enter the password again) for it to work again.

**Comment by BlackDex** [comment]
Might be related to this: https://github.com/bitwarden/clients/issues/9253
If you say that it is working with an older version of the extension, then i think something has changed on the client side which is causing an issue.

You might want to try a clean account and add some test entries and see if that keeps working.
Then if that is the case you can try to export your vault, and import it into the test account and see if it starts failing again.
If that is the case, there might be some entry which is causing issues.

**Comment by BlackDex** [comment]
Also, what KDF settings do you have configured?

**Comment by LiloBzH** [comment]
yes that's this issue !

KDF : 600000

**Comment by bunicb** [comment]
Apparently had it set to the previous default of 10000. Set it to 600000, no change.

Tried it with a clean account and it seems to work. Imported data, works, added it to the organization and collections, and it seems to work fine. Still doesn't work with my original account.

Not sure what the issue is, it looks like the only differences are attachments in some items, which are not present in current items and added Duo 2FA. Will try adding those one by one to see if there are any differences.

**Comment by bunicb** [comment]
Added some attachments, and added Duo integration. Duo doesn't seem to work because it now wants a Universal prompt, but otherwise the new account works, so it seems something is odd with my old account.

**Comment by stiffbruno** [comment]
Hello, I have the same problem since Chrome extension version 2024.5.0. Has anyone found a solution yet?

**Comment by BlackDex** [comment]
There has been a new release of the extensions just a few hours ago which contains fixed for decryption issues. It probably is related to that. Since this does look more like a client issue, going to move this from issues.

**Reply by tessus** [reply]
This is the second or 3rd time you moved the same issue to the discussions. IMO this should stay an issue until this has been resolved. It's not that this is something only one person encounteres. Several people have this issue.  

P.S.: Maybe I mixed this up with another issue. Nevertheless, this doesn't change the fact that this is rather an issue than a discussion.

**Reply by bunicb** [reply]
I don't see it a problem to be a discussion, instead of an issue, as it's not inherently a vaultwarden issue, but a bitwarden client issue. Although for visibility, it would probably be better as an issue.

**Comment by bunicb** [comment]
2024.5.1 didn't fix it, 2024.5.2 seems to work.

**Comment by LiloBzH** [comment]
fix publish for Chrome but not for firefox !

it's a joke ^^
